### PR TITLE
Detect metered connections for macos when connected to an iOS Personal Hotspot

### PR DESCRIPTION
**Pull request by jramnani**
### Description

This change addresses #1884 by adding support for Vorta to detect when macOS is connected to a Wi-Fi network presented by an iOS Personal Hotspot.

I changed how the `darwin` module gets its information about Wi-Fi networks to use ObjC instead of running command line tools.  The benefits of using the ObjC Cocoa APIs is that we get richer information about Wi-Fi networks. There is an explicit API in Cocoa for checking whether a connected Wi-Fi network is an iOS Personal Hotspot. 

I left the existing implementation in place, and added a new check for iOS since it seems like the previous implementation was working when tethered with an Android device.

### Related Issue

* #1884 

### Motivation and Context

I updated this part of the code base because while I was working through an home Internet outage recently, a Vorta backup running in the background used up all of my allotted data on my mobile phone plan.

### How Has This Been Tested?

* Automated tests were added and improved in the `network_manager.darwin` module.
* I ran a manual tests against a remote SSH repository.  I connected to a Personal Hotspot Wi-Fi, then tried to run a backup.  Vorta successfully detected the Hotspot connection and suppressed the backup with a status message for the user.

### Screenshots (if appropriate):

Vorta Success Screenshot Redacted

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] I have read the [CONTRIBUTING](https://vorta.borgbase.com/contributing/) guide.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

*I provide my contribution under the terms of the [license](./../LICENSE.txt) of this repository and I affirm the [Developer Certificate of Origin][dco].*

[dco]: https://developercertificate.org/

### CODE REVIEW

**Comment by m3nu** [comment]
Looks good in principle, but it fails when there is no Wifi at all. This case should be covered.

```
        wifis = []
        interface: CWInterface = self._get_wifi_interface()
>       interface_name = interface.name()
E       AttributeError: 'NoneType' object has no attribute 'name'
```

**Comment by jramnani** [comment]
I'm glad to see the test suite caught this.

I appreciate that I could limit my changes to one module.  This change was easier to do because of that. Thank you! :)

**Comment by m3nu** [comment]
One minor thing left: It doesn't like the order or imports.

**Comment by jramnani** [comment]
Fixed. Sorry about that. I had assumed PyCharm would add the import for me correctly.

Now I know to run `make lint` locally to make sure.

File: src/vorta/network_status/darwin.py [hunk 1]
```diff
@@ -1,53 +1,82 @@
 import subprocess
 from datetime import datetime as dt
-from typing import Iterator, Optional
+from typing import Iterator, List, Optional
+
+from CoreWLAN import CWInterface, CWNetwork, CWWiFiClient
```
Comment (real-yfprojects) [inline src/vorta/network_status/darwin.py:5]: I was wondering whether this toplevel import is a good practice since the dependency is optional.

Comment (jramnani) [reply src/vorta/network_status/darwin.py:5]: Are you suggestion suppressing import errors like what's happening in the `notifications` module?

https://github.com/borgbase/vorta/blob/master/src/vorta/notifications.py#L8-L11

I have a Linux host I can test with.  I'll see if I can do that soon to see if the application throws an ImportError for me.

Comment (jramnani) [reply src/vorta/network_status/darwin.py:5]: I can report that on Pop_OS 22.04 I was able to start the app in development mode and do a successful backup to a remote host.  Also the full test matrix passed for me via `make test-unit` on the same host.

I saw an INFO log message: `Using NetworkManagerMonitor NetworkStatusMonitor implementation`

No INFO or DEBUG logs mentioning the new imports.

I do see some DEBUG log messages from `vorta.keyring.abc` about `No module named 'objc'`, but they have not affected the manual backup test.

I'm happy to make the change if we want it.  But wanted to report back these manual testing results.

Comment (jramnani) [reply src/vorta/network_status/darwin.py:5]: An upshot of my local testing was that I also tried a backup while tethered to my iOS device and it detected the metered connection. 🎉

**Review by m3nu — APPROVED** [review]

**Pull request merged**